TestUtilities/Helpers/DDAssert.swift

```diff
@@ -10,8 +10,11 @@
 import Foundation
 import XCTest
 
-private enum DDAssertError: Error {
-    case expectedFailure(String, keyPath: [String] = [])
+public enum DDAssertError: Error {
+    /// An error indicating known coding failure that happened in evaluated `expression()` block.
+    case expectedCodingFailure(String, keyPath: [String])
```

Comment:
It's not necessary a coding failure, is it? we also provide a keyPath for reflection and dictionary comparison. 
The previous `expectedFailure` with optional keyPath already covered both cases IMO. But maybe it wasn't good for auto-completion?

Reply:
I reverted this change 👍. My concern was that this concept doesn't scale well if we want to provide other detail than "no" or "`keyPath: [String] = []`" (we'll have to introduce another associated value with default). However, we don't need it for now, so reverting makes sense - we can revisit it later.